x/superfluid/keeper/msg_server_test.go

```diff
@@ -192,3 +193,191 @@ func (suite *KeeperTestSuite) TestMsgLockAndSuperfluidDelegate() {
 		}
 	}
 }
+
+// TestMsgSuperfluidDelegate_Event tests that events are correctly emitted
+// when calling SuperfluidDelegate.
+func (suite *KeeperTestSuite) TestMsgSuperfluidDelegate_Event() {
+	type param struct {
+		lockOwner sdk.AccAddress
+		duration  time.Duration
+	}
+
+	tests := []struct {
+		name       string
+		param      param
+		expectPass bool
+	}{
+		{
+			name: "basic valid",
+			param: param{
+				lockOwner: sdk.AccAddress([]byte("addr1---------------")), // setup wallet
+				duration:  time.Hour * 504,
+			},
+			expectPass: true,
+		},
+		{
+			name: "invalid",
```

Comment:
can we be more specific just for future debugging, something like "invalid due to duration below minimum" or something along those lines?